api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -118,12 +118,19 @@ type FelixConfigurationSpec struct {
 	// attempts to acquire the iptables lock if it is not available. Lower values make Felix more
 	// responsive when the lock is contended, but use more CPU. [Default: 50ms]
 	IptablesLockProbeInterval *metav1.Duration `json:"iptablesLockProbeInterval,omitempty" configv1timescale:"milliseconds" confignamev1:"IptablesLockProbeIntervalMillis"`
-	// FeatureDetectOverride is used to override the feature detection.
+	// FeatureDetectOverride is used to override the environment feature detection.
 	// Values are specified in a comma separated list with no spaces, example;
 	// "SNATFullyRandom=true,MASQFullyRandom=false,RestoreSupportsLock=".
 	// "true" or "false" will force the feature, empty or omitted values are
 	// auto-detected.
 	FeatureDetectOverride string `json:"featureDetectOverride,omitempty" validate:"omitempty,keyValueList"`
+	// FeatureGate is used to enable or disable Calico features.
+	// Values are specified in a comma separated list with no spaces, example;
+	// "BPFConnectTimeLoadBalancingWorkaround=enabled,XyZ=false". This is
+	// primarily used to enable feature that is not fully production ready or
+	// helps to fix a corner case in certain environments, but is not need most
+	// of the time.
+	FeatureGate string `json:"featureGate,omitempty" validate:"omitempty,keyValueList"`
```

Comment:
nit, but I think it should be FeatureGates (plural) since there are multiple independent gates that we expose.

Given this is in the v3 API, IMO it should really be a slice format rather than forcing a comma-separated list, which is hard for users to work with. Not a deal-breaker, but if it's not too much of a pain this would be nicer:

```
featureGates:
- name: BPFConnectTimeLoadBalancingWorkaround
  value: false

Reply:
I understand we already have a bunch of other slices in this struct represented as comma-separated values, so we don't need to block on this change. It's just too bad that we've sort of built that convention which is a bit awkward comparatively.

Reply:
(just ignore the switch to a slice, but I do think the name should be plural)